src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs

```diff
@@ -420,31 +420,32 @@ private async Task StartAsyncCore(CancellationToken cancellationToken)
             CheckDisposed();
 
             Log.Starting(_logger);
-
-            // Start the connection
-            var connection = await _connectionFactory.ConnectAsync(_protocol.TransferFormat);
-            var startingConnectionState = new ConnectionState(connection, this);
-
-            // From here on, if an error occurs we need to shut down the connection because
-            // we still own it.
-            try
-            {
-                Log.HubProtocol(_logger, _protocol.Name, _protocol.Version);
-                await HandshakeAsync(startingConnectionState, cancellationToken);
-            }
-            catch (Exception ex)
+            using (var cts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken, _state.StopCts.Token))
```

Comment:
I think that it's better to have both callers of StartAsyncCore, StartAsyncInner and ReconnectAsync, should explicitly pass in _state.StopCts.Token. So instead of creating the linked CTS here, you would do so in StartAsyncInner. In ReconnectAsync, you can just pass _state.StopCts.Token instead of CancellationToken.None.

This change would mean we would no longer have to link _state.StopCts.Token inside of HandshakeAsync.

Reply:
Refactored by commit 62ea869